src/SDK/Common/Configuration/Configuration.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Configuration;
+
+/**
+ * Configuration can come from one or more of the following sources (from highest to lowest priority):
+ * - values defined in php.ini
+ * - environment variable
+ * - configuration file (todo)
+ */
+class Configuration
+{
+    private static ?array $resolvers = null;
+
+    /**
+     * @return array<Resolver>
+     */
+    private static function resolvers(): array
+    {
+        self::$resolvers ??= [
+            new IniResolver(),
+            new EnvironmentResolver(),
+        ];
+
+        return self::$resolvers;
+    }
+
+    public static function has(string $name): bool
+    {
+        foreach (self::resolvers() as $resolver) {
+            if ($resolver->hasVariable($name)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static function getResolver(string $name): Resolver
```

Comment:
Encapsulate this in a `CompositeResolver` and provide the composite resolver to `Accessor::getXxx()` to get rid of the `self::getResolver()` calls?

Reply:
I did something similar, but I thought it looked neater if `Accessor` just referenced the `CompositeResolver` instance (which is closer to the original code)... wdyt?

Reply:
I'm fine with either solution; we should also move the `Accessor` logic into `Configuration` as they do pretty much the same now. (This is the change that I had first in mind; I opted for the `CompositeResolver` argument because it improves testability a bit.)